cli/silence_update.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/prometheus/alertmanager/cli/format"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type getResponse struct {
+	Status    string        `json:"status"`
+	Data      types.Silence `json:"data,omitempty"`
+	ErrorType string        `json:"errorType,omitempty"`
+	Error     string        `json:"error,omitempty"`
+}
+
+var updateFlags *flag.FlagSet
+var updateCmd = &cobra.Command{
+	Use:     "update <id> ...",
+	Aliases: []string{"extend"},
+	Args:    cobra.MinimumNArgs(1),
+	Short:   "Update silences",
+	Long:    `Extend or update existing silence in Alertmanager.`,
+	Run:     CommandWrapper(update),
+}
+
+func init() {
+	updateCmd.Flags().StringP("expires", "e", "", "Duration of silence (100h)")
+	updateCmd.Flags().String("expire-on", "", "Expire at a certain time (Overwrites expires) RFC3339 format 2006-01-02T15:04:05Z07:00")
```

Comment:
In alertmanager, you can also update the starting time when editing a silence.

How about adding that as well, and naming the flags `date.start` and `date.end`?

Reply:
hm, thinking more about the name .. not sure i like `date`. @grobie , naming guru, any recommendation?

Reply:
Should we just follow the field names in Alertmanager API, `startsAt` and `endsAt`?

Reply:
hrm, those are more json-idiomatic names. for a CLI I think something different would be more appropriate.